sklearn/preprocessing/tests/test_target.py

```diff
@@ -0,0 +1,263 @@
+
+
+def test_transform_target_regressor_functions():
+    X, y = friedman
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      func=np.log, inverse_func=np.exp)
+    y_pred = regr.fit(X, y).predict(X)
+    # check the transformer output
+    y_tran = regr.transformer_.transform(y.reshape(-1, 1)).squeeze()
+    assert_allclose(np.log(y), y_tran)
+    assert_allclose(y, regr.transformer_.inverse_transform(
+        y_tran.reshape(-1, 1)).squeeze())
+    assert_equal(y.shape, y_pred.shape)
+    assert_allclose(y_pred, regr.inverse_func(regr.regressor_.predict(X)))
+    # check the regressor output
+    lr = LinearRegression().fit(X, regr.func(y))
+    assert_allclose(regr.regressor_.coef_.ravel(), lr.coef_.ravel())
+
+
+def test_transform_target_regressor_functions_multioutput():
+    X = friedman[0]
+    y = np.vstack((friedman[1], friedman[1] ** 2 + 1)).T
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      func=np.log, inverse_func=np.exp)
+    y_pred = regr.fit(X, y).predict(X)
+    # check the transformer output
+    y_tran = regr.transformer_.transform(y)
+    assert_allclose(np.log(y), y_tran)
+    assert_allclose(y, regr.transformer_.inverse_transform(y_tran))
+    assert_equal(y.shape, y_pred.shape)
+    assert_allclose(y_pred, regr.inverse_func(regr.regressor_.predict(X)))
+    # check the regressor output
+    lr = LinearRegression().fit(X, regr.func(y))
+    assert_allclose(regr.regressor_.coef_.ravel(), lr.coef_.ravel())
+
+
+def test_transform_target_regressor_1d_transformer():
+    # All transformer in scikit-learn expect 2D data. FunctionTransformer with
+    # validate=False lift this constraint without checking that the input is a
+    # 2D vector. This test check the consistency of the data shape using this
+    # transformer and a 1D y array
+    X, y = friedman
+    transformer = FunctionTransformer(func=lambda x: x + 1,
+                                      inverse_func=lambda x: x - 1,
+                                      validate=False)
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      transformer=transformer)
+    y_pred = regr.fit(X, y).predict(X)
+    assert_equal(y.shape, y_pred.shape)
+    # consistency forward transform
+    y_tran = regr.transformer_.transform(y)
+    _check_shifted_by_one(y, y_tran)
+    assert_equal(y.shape, y_pred.shape)
+    # consistency inverse transform
+    assert_allclose(y, regr.transformer_.inverse_transform(
+        y_tran).squeeze())
+    # consistency of the regressor
+    lr = LinearRegression()
+    transformer2 = clone(transformer)
+    lr.fit(X, transformer2.fit_transform(y))
+    y_lr_pred = lr.predict(X)
+    assert_allclose(y_pred, transformer2.inverse_transform(y_lr_pred))
+    assert_allclose(regr.regressor_.coef_, lr.coef_)
+
+
+def test_transform_target_regressor_1d_transformer_multioutput():
```

Comment:
It is hard to see how similar or different the code is here from the previous test. Perhaps use a loop or or a check function or `pytest.mark.parametrize`